tests/testBootprint.py

```diff
@@ -0,0 +1,246 @@
+# the GNU General Public License along with this program.  If not,
+# see <https://www.lsstcorp.org/LegalNotices/>.
+#
+
+from __future__ import absolute_import, division, print_function
+import unittest
+import tempfile
+
+import lsst.utils.tests
+import lsst.afw.geom as afwGeom
+import lsst.afw.detection as afwDet
+import lsst.afw.table as afwTable
+
+
+class BootprintTestCase(unittest.TestCase):
+    def setUp(self):
+        self.spanRad = 4
+        self.regionRad = 10
+        self.spans = afwGeom.SpanSet.spanSetFromShape(self.spanRad,
+                                                      afwGeom.Stencil.BOX)
+        minPoint = afwGeom.Point2I(-self.regionRad, -self.regionRad)
+        maxPoint = afwGeom.Point2I(self.regionRad, self.regionRad)
+        self.region = afwGeom.Box2I(minPoint, maxPoint)
+        self.schema = afwDet.PeakTable.makeMinimalSchema()
+        # Run the the constructors test to ensure the Bootprints are setUp
+        self.testConstructors()
+
+    def tearDown(self):
+        del self.spans
+        del self.region
+        del self.schema
+        del self.footprint
+        del self.footprintWithRegion
+        del self.footprintWithSchema
+        del self.footprintWithSchemaRegion
+
+    def testConstructors(self):
+        '''
+        Test that each of the constructors constructs a valid Bootprint,
+        if any of these fails, an exception will be raised and the test
+        will fail.
+        '''
+        self.footprint = afwDet.Bootprint(self.spans)
+        self.footprintWithRegion = afwDet.Bootprint(self.spans, self.region)
+        self.footprintWithSchema = afwDet.Bootprint(self.spans, self.schema)
+        self.footprintWithSchemaRegion = afwDet.Bootprint(self.spans,
+                                                          self.schema,
+                                                          self.region)
+
+    def testIsHeavy(self):
+        self.assertFalse(self.footprint.isHeavy)
+
+    def testGetSetSpans(self):
+        '''
+        Test getting and setting the SpanSet member of the Bootprint with both
+        the getters and setters and the python property accessor
+        '''
+        self.assertTrue(self.footprint.getSpans() == self.spans)
+        self.assertTrue(self.footprint.spans == self.spans)
+        tempSpanSet = afwGeom.SpanSet.spanSetFromShape(2, afwGeom.Stencil.BOX)
+        self.footprint.setSpans(tempSpanSet)
+        self.assertTrue(self.footprint.spans == tempSpanSet)
+        # reset back to origional with property
+        self.footprint.spans = self.spans
+        self.assertTrue(self.footprint.spans == self.spans)
+
+    def testPeakFunctionality(self):
+        newSchema = afwDet.PeakTable.makeMinimalSchema()
+        newField = afwTable.Field_I("extra", "doc", "na")
+        newSchema.addField(newField)
+        self.footprint.setPeakSchema(newSchema)
+        names = self.footprint.getPeaks().getSchema().getNames()
+        self.assertTrue("extra" in names)
```

Comment:
I talked with Fred about this, and there seemed to be no single place to look at all the assert statements, so in lie of wasting more time, I just coded something that worked.